File: engine/src/main/java/ru/vertuos/engine/hedgehog/listener/HedgehogEventListener.java
package ru.vertuos.engine.hedgehog.listener;

public interface HedgehogEventListener {

    void onGameOver();

    void onDie(int livesLeft);

    void onTakeDamage(float damage);

    void onRecover(float recoverValue);

    void onLossRings();
}

File: engine/src/main/kotlin/ru/vertuos/engine/hedgehog/listener/HedgehogEventListener.kt
package ru.vertuos.engine.hedgehog.listener

interface HedgehogEventListener {

    fun onGameOver()

    fun onDie(livesLeft: Int)

    fun onTakeDamage(damage: Float)

    fun onRecover(recoverValue: Float)

    fun onLossRings()
}
